### PR TITLE
Refactored lookups and mixins

### PR DESCRIPTION
**Pull request by peterfarrell**
Refactored lookups and mixins to not be so redundant in the terms of classes and simplified SQL snippets in `__init__.py`

### CODE REVIEW

**Comment by peterfarrell** [comment]
@kevinetienne @meshy @maxpeterson @cyanc A review of this PR would be help considering the amount of refactoring I did. Would anyone volunteer?

**Comment by maxpeterson** [comment]
I am not very familiar with the code but it looks good to me, and seems like a better approach

File: pgcrypto/apps.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from django.apps import AppConfig
+
+
+class PgcryptoConfig(AppConfig):
+    name = 'pgcrypto'
+
+    def ready(self):
+        """Ready for pgcrypto."""
+        pass
```
[the remark this answers is not in the document]
Comment (peterfarrell) [reply pgcrypto/apps.py]: Yes, you are right -- we aren't doing any signals in this library. I will remove that and also add  a `verbose_name`.

File: pgcrypto/__init__.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 from django.conf import settings
 
+default_app_config = 'pgcrypto.apps.PgcryptoConfig'
```
[the remark this answers is not in the document]
Comment (peterfarrell) [reply pgcrypto/__init__.py]: Yeah I thought about that but decided to not introduce any more backwards compatibility / upgrade issues. I don't have strong thoughts either way in the end. I will however use the new approach and write an how to upgrade from Django-PGCrypto-Fields 2.2 to 2.3 in the README.

File: pgcrypto/apps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (meshy) [reply pgcrypto/apps.py]: Well, I'm not sure we need the app config at all! It's just adding the defaults.

**Pull request merged**